### ISSUE TITLE
MySQL 8: FULLTEXT indexes can cause runtime panic

### ISSUE THREAD

**Issue by s123121** [skeema/skeema#106]
Hello, I want to evaluate skeema for my project however I can not run even the init command as show in the getting started guide
```
skeema init -h my.prod.db.hostname  -u bluework -p -d schemas
```
Here is the error
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1337c37]

goroutine 55 [running]:
github.com/skeema/tengo.(*Index).Definition(0x0, 0x1, 0x8, 0x0, 0xc000083440, 0x29)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/index.go:24 +0x37
github.com/skeema/tengo.(*Table).GeneratedCreateStatement(0xc0002c6700, 0x1, 0x8, 0x0, 0x2615, 0x0)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/table.go:52 +0x21e
github.com/skeema/tengo.(*Instance).querySchemaTables.func1(0x8, 0x14b4910)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/instance.go:1001 +0x1b9
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0002b0690, 0xc00006cc40)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
kimlam@MacBook-Pro-2:~/dev/blue-work/server$ skeema init -h localhost -u bluework -p -d schemasEnter password: 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1337c37]

goroutine 51 [running]:
github.com/skeema/tengo.(*Index).Definition(0x0, 0x1, 0x8, 0x0, 0xc000086840, 0x29)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/index.go:24 +0x37
github.com/skeema/tengo.(*Table).GeneratedCreateStatement(0xc00026c8c0, 0x1, 0x8, 0x0, 0x126b4, 0x0)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/table.go:52 +0x21e
github.com/skeema/tengo.(*Instance).querySchemaTables.func1(0x8, 0x14b4910)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/github.com/skeema/tengo/instance.go:1001 +0x1b9
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0001e55f0, 0xc0000b1700)
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/travis/gopath/src/github.com/skeema/skeema/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```
I am using mysql  Ver 8.0.17 for osx10.14 on x86_64 (Homebrew)

**Comment by evanelias** [comment]
Thank you for the issue report. This is the first time I've seen this one -- seems somehow an index is being introspected incorrectly.

A few initial questions to help debug this:
* Just to confirm, you're running MySQL on MacOS natively via Homebrew? (as opposed to in a Docker container)
* Do any of your databases or tables have names including uppercase characters, or are they all entirely lowercase?
* What version of Skeema are you running? (output of `skeema version`)

Thanks!

**Comment by evanelias** [comment]
One more question -- do you use any FULLTEXT indexes? I just spotted a potential Skeema bug regarding them specifically affecting MySQL 8 only.

**Comment by s123121** [comment]
Thanks for the fast reply. 

> Just to confirm, you're running the MySQL server on MacOS natively via Homebrew? (as opposed to in a Docker container)

Yes, mysql Ver 8.0.17 installed via homebrew
> Do any of your databases or tables have names including uppercase characters, or are they all entirely lowercase?

Unfortunately yes, i use camelCase naming for my tables
> What version of Skeema are you running? (output of skeema version)

skeema version 1.3.0, commit 11cfcc8c4540c566d8bee4bd153e588b11906d88, released 2019-09-06T17:03:02Z
> -- do you use any FULLTEXT indexes?

I have a FULLTEXT index in one of my table which includes many columns

**Comment by evanelias** [comment]
Awesome, thank you, I've narrowed it down to a bug in Skeema's FULLTEXT index support affecting MySQL 8 and Percona Server 8. Skeema's support for FULLTEXT indexes is pretty recent (just added a month ago) so still uncovering some edge cases. I'll aim to have a patch release with a fix within the next day, and will comment on here once that's ready. Sorry about the bug and thank you again for the report! 

Separately, earlier this year I received a report of issues with mixed-case table names and db names when the database server is running natively on MacOS, due to MacOS having a case-insensitive filesystem and this affects MySQL's behavior in subtle ways. So it's possible you may also encounter Skeema problems with the camelCase table names, but let's figure that out after the FULLTEXT bug in Skeema is fixed.

**Comment by evanelias** [comment]
Fix merged to master and included in new patch release [v1.3.1](https://github.com/skeema/skeema/releases/tag/v1.3.1). Thanks again for the issue report!

Separately please let me know if you encounter anything weird from the mixed-case identifiers when running the db server on MacOS. One user previously reported on MacOS that `skeema init` worked fine, but subsequent calls to `skeema diff` would incorrectly generate a diff containing all contents of the *.sql files. The root cause (in that case) seemed to relate to mixed case *database* names in a case-insensitive filesystem, likely due to some subtlety of how this affects the cases returned in MySQL's `information_schema` vs Skeema tracking relational data in a manner that assumes case-sensitive identifiers. The previously-reported problem is described in #65. Not clear yet if mixed-case *table* names cause a similar problem with a MacOS db server.
